### DISCUSSION TITLE
Cannot create a new record with SQL Server (Error: "ID can't be blank")

### DISCUSSION THREAD

**Discussion by egattulli** [Budibase/budibase#10299] [Help] [answered]
Hi everyone!

I designed a form that should create a new record in a table located in SQL Server. When I try to create a new record, I get this error message: "ID can't be blank". From what I understand, this has to be related to the ID field in the table that is a primary key and auto increment. I cannot insert this text field in the form, so I cannot even try to manually write the ID. For the purpose of this database, I need to keep the ID field.

Does anyone know how to solve this issue?

Thanks in advance.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @egattulli,

I started from the beginning here and created a table using the following code:

```
CREATE TABLE pets (
    id INT NOT NULL AUTO_INCREMENT,
    name VARCHAR(255) NOT NULL,
    type VARCHAR(255) NOT NULL,
    breed VARCHAR(255),
    age INT,
    PRIMARY KEY (id)
);
```

I then imported this table into Budibase through the adding of a datasource MySQL.
<img width="1338" alt="Screenshot 2023-04-14 at 11 05 59" src="https://user-images.githubusercontent.com/126772285/232016890-124222f7-f635-4dd9-b162-9b9ffe73542c.png">

<img width="657" alt="Screenshot 2023-04-14 at 11 11 07" src="https://user-images.githubusercontent.com/126772285/232016822-0a21068b-b7d2-44ad-ba9e-f9a1d92d345c.png">

**End result below,** I was able to create the row and the auto increments of the id works as intended.
<img width="993" alt="Screenshot 2023-04-14 at 11 11 42" src="https://user-images.githubusercontent.com/126772285/232017030-234acd7d-2300-4751-9d94-8abdaf94f047.png">

I think your issue may be how you created the table initially with sql. Could you try recreating it again, importing and then try to create a row?

Hopefully this helps, if you still can't figure out the issue. Could you link me your sql schema or db dump and ill have a quick look at that.

**Reply by egattulli** [reply]
Thank you @ConorWebb96, now it works! I had to recreate the table and fetch it in Budibase.
